### ISSUE TITLE
Cannot select item from dropdown list with optgroups

### ISSUE THREAD

**Issue by jaapterwoerds** [cypress-io/cypress#367]
**Description**

I cannot select an item from a list when the list uses optgroups to separate groups of items.

**Steps To Reproduce**

I have a page 'index.html' with a dropdown list called 'cars' with two optgroups:
```
<!DOCTYPE html>
<html>
  <body>
    <select name="cars">
      <optgroup label="Swedish Cars">
        <option value="volvo">Volvo</option>
        <option value="saab">Saab</option>
      </optgroup>
      <optgroup label="German Cars">
        <option value="mercedes">Mercedes</option>
        <option value="audi">Audi</option>
      </optgroup>
    </select>
  </body>
</html>
```
When I try to select 'Volvo' I as follows:
```
cy
      .visit('/index.html')
      .get('[name="cars"]').select('Volvo')
```

I get an error: 
```
Command:     select
runner.js:62350 Applied to:  <select name=​"cars">​…​</select>​
runner.js:62350 Error:       CypressError: Timed out retrying: cy.select() failed because it could not find a single <option> with value or text matching: 'Volvo'
```
When I remove the optgroups, the selection succeeds.

I would expect I don't need any special syntax to select an item from a dropdown list which has optgroups.

**Comment by jaapterwoerds** [comment]
If there is a workaround for this issue than I would be happy to hear 😄

**Comment by jennifer-mann** [comment]
Cypress `.select()` is only traversing one DOM level down to find available `options`. The code needs to be updated to account for `optgroups`. 

Using `.click()` on `<select>` elements currently throws the error 

```
cy.click() cannot be called on a <select> element. 
Use cy.select() command instead to change the value.
```

So, the way to workaround this would be to use plain ol' JavaScript & jQuery:

```
cy.get('select').then(function($select){
  $select.val('volvo')
})
cy.get('select').should('have.value', 'volvo')
```

**Comment by brian-mann** [comment]
Fixed in [`0.19.1`](https://github.com/cypress-io/cypress/wiki/Changelog#0191-03092017).

**Comment by jaapterwoerds** [comment]
Awesome, thx!
